### ISSUE TITLE
Feature suggestions

### ISSUE THREAD

**Issue by chaoscreater** [codecoding/PluralSightDownloader#21]
Hi there!

Really love your work and so glad I came across this! Have been looking for a tool like this for ages and this has definitely made it easier to download the videos offline. Thanks for sharing!

While the app is awesome, I do have a few suggestions to make.

1. Let it remember your username and password instead of asking you every time you launch it.

2. Stop the videos from playing when you download a course.

3. Allow some sort of quick navigation bar to jump to the home page. Or, better yet, implement a search bar so we can search for videos/courses easier.

4. Allow you to clear the course list in the download panel on the right side. Every time I've finished downloading a course, I'll have to hit the backspace button on my keyboard to go back to the homepage (so that I can search for other videos to download). By default, the videos start playing when you download. So when you go back a page, it goes back to the course page, which triggers the app to generate a download on the right hand side. The more courses you download, the more cluttered the right hand panel becomes. It also auto expands so you have to constantly collapse the download-list in the right panel.

5. Remember saved video location.

6. Allow batch downloads into a folder and automatically naming each download by its course name. Not sure how this will be implemented, but I do know that you can bookmark a course and also create playlists and add courses to it. Perhaps if your app can detect a playlist (which we can just download all in one click of a button), that would be awesome.

**Comment by robertohuertasm** [comment]
Hi @chaoscreater,

thank you for your amazing feedback. I take note of all the points but I don't know if I'll have the time to implement anything as I'm a little bit busy lately. Anyway, if you or anyone else want to collaborate, you'll be very welcome!!

Best regards,
Roberto.

**Comment by chaoscreater** [comment]
I would love to if I knew how to code :(

But yeah no rush, just thought I might add a few suggestions in case someone wants to pick it up :D

**Comment by Sonic3R** [comment]
Hi,

I changed code for myself to be easy for me to download videos.

1. I don't know if it is secure to remember password (need to catch some cookies and store somewhere in the file). The downloader uses CefSharp which is a separate process. If you close application, automatically will shut down the CefSharp process where is stored cookies and local storage.

2. It is a local storage setting set in browser which says to not autoplay videos.

3. I put BACK button to be easy for me.

4. I changed code in order to refresh tree with new course details (modules and clips). I changed that will be collapsed all nodes except the root node.

5. It is not ok .... Depends on each desire to download separate folder or not. Maybe needs a checkbox like "Remember location".

6. I changed code in order to save in specific folder and each module in separate folder like 1. Introduction    2. Something else, etc.

But somehow, I don't know the reason, first module contains a clip which VideoUrl comes with an error: "The custom error module does not recognize this error."

Happens always to first module, no matter what name is... Strange.
